vector/src/main/java/im/vector/app/features/MainActivity.kt

```diff
@@ -103,15 +116,36 @@ class MainActivity : VectorBaseActivity<ActivityMainBinding>(), UnlockedActivity
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        args = parseArgs()
-        if (args.clearCredentials || args.isUserLoggedOut || args.clearCache) {
-            clearNotifications()
+
+        mainViewModel.viewEvents.stream()
+                .onEach(::handleViewEvents)
+                .launchIn(lifecycleScope)
+
+        mainViewModel.handle(MainViewAction.StartApp)
+    }
+
+    private fun handleViewEvents(event: MainViewEvent) {
+        when (event) {
+            MainViewEvent.AppStarted -> handleAppStarted()
         }
-        // Handle some wanted cleanup
-        if (args.clearCache || args.clearCredentials) {
-            doCleanUp()
+    }
+
+    private fun handleAppStarted() {
+        if (intent.hasExtra(EXTRA_NEXT_INTENT)) {
+            // Start the next Activity
+            val nextIntent = intent.getParcelableExtra<Intent>(EXTRA_NEXT_INTENT)
+            startIntentAndFinish(nextIntent)
```

Comment:
this is working but when going back, we can see the MainActivity screen for a few ms...